### ISSUE TITLE
[BUG] calling Fortran subroutine from Python, but the wrong output 

### ISSUE THREAD

**Issue by skphy** [rjfarmer/gfort2py#39]
**Describe the bug**
calling Fortran subroutine  from Python, but the wrong output is returned

**Fortran code**
	
	module module_data
	   implicit none
	   public :: datamu
	contains
	   subroutine datamu(y, xp, yp)
		  use iso_fortran_env, only: dp => real64

		  real(dp), intent(inout) :: y
		  real(dp), intent(inout) :: xp(:)
		  real(dp), intent(inout) :: yp(:)

		  y = -1000.0_dp
		  xp = [13.0_dp, -2.0_dp]
		  yp = [1.0_dp, -42.014_dp]
		  print *, "inside datamu sub: y, xp, yp", y, xp, yp
	   end subroutine datamu
	end module module_data

**Python code**

	import numpy as np
	import gfort2py as gf

	SHARED_LIB_NAME="./module_data.dll"
	MOD_FILE_NAME="./module_data.mod"

	pes = gf.fFort(SHARED_LIB_NAME,MOD_FILE_NAME)

	y = 0.
	x = np.array([-0.6, 1.25]) 
	yp = np.array([0.,0.])   

	output = pes.datamu(y, x, yp) 
	print('output: {}'.format(output))

	print("NOTE: output[1]  also different from the output part (see output above)")
	print(output[0])
	print(output[1])

**Software (please complete the following information):**
 - OS: Linux
 - gfort2py version:: '2.2.1'

 - gfortran version: 9.4.0
 - gfortran compile options: gfortran -fPIC -shared -o module_data.dll module_data.f90 
 - Python version: Python 3.9.12
 - If this involves arrays numpy version: '1.22.3'

**Additional context**
	output from the above program: 
	 inside datamu sub: y, xp, yp  -1000.0000000000000        13.000000000000000       -2.0000000000000000        1.0000000000000000       -42.014000000000003     
	output: Result(result=None, args={'y': -1000.0, 'xp': array([13., -2.]), 'yp': array([13.,  2.])})
	None
	{'y': -1000.0, 'xp': array([13.,  2.]), 'yp': array([13., -2.])}
	So, I expect the output similar to what is inside the datamu subroutine (output precision truncated), i.e. y = -1000.00000, xp = [13.00,  -2.00], and  yp= [1.0000, -42.014]. 

	**In a nutshell, the output yp is not correct.**

**Comment by skphy** [comment]
thanks, it works now!!
